### PR TITLE
added outputConsumer to StreamedContent

### LINKED ISSUE #7233: FileDownload: Write direct to the Outputstream
**Description**
In many applications some data, e.g. csv file will be generated on the fly. so it would be a memory efficent feature to write directly to the outputstream

**Describe the solution you would like**
Maybe a vistor pattern could be the solution, so you can use the stream as is or define a outputstream consumer.

```
 
			return DefaultStreamedContent.builder().contentType("text/csv").name("my.csv")
					.outputStream((outputstream) -> { 
//do something with the outputstream
						}).build();

```

### PR DESCRIPTION
**Pull request by neXus1987**
this allows us to write directly to the responseOutputStream

### CODE REVIEW

**Comment by melloware** [comment]
Can you do a bare minimum PR without any formatting changes?  I see a lot of code has been reformatted.

**Comment by neXus1987** [comment]
sure this should be better

**Comment by tandraschko** [comment]
i think we may also check this for GraphicImage
also please add this to the docs + add some javadoc in StreamedContent

**Comment by Rapster** [comment]
Is it possible to provide a sample of your use case?

**Comment by neXus1987** [comment]
this is my use case:

```
        DefaultStreamedContent.builder().contentEncoding("text/csv").name("my.csv").outputConsumer((os)->{
            try (PrintWriter p = new PrintWriter(new OutputStreamWriter(os, "UTF-8"))) {
             
                    p.println("test-line");
              
            } catch (Exception e1) {
                e1.printStackTrace();
            }
        }).build();
```

**Comment by Rapster** [comment]
@neXus1987 If you're using omnifaces, you can use `Faces#sendFile` which will do exactly what you want and you don't need `p:fileDownload`

**Comment by tandraschko** [comment]
the change makes totally sense for me as we dont need to buffer
however, not sure about the name

maybe just "writer"?
`Consumer<OutputStream> writer`

**Comment by Rapster** [comment]
I was more thinking something like `outputCallback`, but up to you writer sounds good too. Otherwise, why using a fileDownload component over something like `Faces#sendFile`?

**Comment by Rapster** [comment]
Ah maybe for the new ajax feature?

**Comment by tandraschko** [comment]
maybe yep
it also makes sense for normal graphicImage streaming as we dont need to buffer anymore

**Comment by Rapster** [comment]
Yeah I see how useful it is for graphicImage, but not sure regarding fileDownload

File: src/main/java/org/primefaces/model/DefaultStreamedContent.java [hunk 1]
```diff
@@ -110,6 +119,11 @@ public Builder contentLength(Integer contentLength) {
             return this;
         }
 
+        public Builder outputConsumer(Consumer<OutputStream> outputConsumer) {
+            streamedContent.outputConsumer = outputConsumer;
+            return this;
+        }
+
         public DefaultStreamedContent build() {
             return streamedContent;
```
Comment (Rapster) [inline src/main/java/org/primefaces/model/DefaultStreamedContent.java:128]: Throws exception if both stream and consumer are not null

**Comment by tandraschko** [comment]
finished by #7295 and #7233 now

**Comment by tandraschko** [comment]
thanks to @neXus1987 for his pre-work

**Pull request closed without merging**